supporting_files/ArrayExpressSiteConfig.yml

```diff
@@ -78,6 +78,7 @@ VALIDATION_SCRIPT:
 # Location of ADF and Experiments checked in Atlas.pm
 ADF_CHECKED_LIST:        /nfs/production3/ma/home/atlas3-production/sw/configs/adfs_not_in_atlas.txt
 ATLAS_EXPT_CHECKED_LIST: /nfs/production3/ma/home/atlas3-production/sw/configs/expts_checked_for_atlas.txt
+USE_CHECKED_LIST_FILE:  false
```

Comment:
In general, and specially when we edit stuff that we don't have entire clarity on where it is being executed, you should always aim to keep the default case to the previous behaviour in place, so that in all those places where this is running, we don't need to change anything. What if this variable is not defined in the supporting file? How will this behave? Maybe change this to `AVOID_CHECKED_LIST_FILES` instead, and consider it false if it is not defined? That would keep the default behaviour intact. I would still avoid adding stuff to config files and prefer setting flags on runtime, since this is the most explicit way of doing it for the user (otherwise, this is buried somewhere in a file that you need to be aware of), but lets just change the default behaviour and merge this as proposed.

Reply:
Thanks @pcm32, good point. Not having the variable should result in the normal behaviour, requiring the files. I will reverse the logic. 

Yes, flags are good if the behaviour dramatically changes and the user needs to be aware of that. To my knowledge nobody is using these files that are being written to here (I didn't even know they existed and for what purpose until we created the conda package). Curators are running this module as part of the validation several times a day every day. I'd really appreciate not having to add another flag every time. So therefore the config change is the better option here with the usecase for curators in mind.

Reply:
Thanks for making the change. I don't think that the length of arguments is a good design constraint, specially for deciding to bury preferences in files; if you look at most unix utilities, they can have a few tenths sometimes easily. If your concern is having to type the command, I would suggest that you set either bash aliases or functions for your calls for increased productivity. For instance:

```
alias getXShellOnFarm='bsub -Is -XF -q q74 $SHELL'
# or a function if you want to use positional arguments
function getShellOnProdFarmWithMem(){
   echo "bsub -Is -q q74 -R 'rusage[mem=$1]' -M $1 $SHELL"
   bsub -Is -q q74 -R \"rusage[mem=$1]\" -M $1 $SHELL
}
```

Reply:
Yes, if we would be writing this from scratch, making it a command line option would be the choice. Another issue here is that the script that would need the additional option, is not even part of the modules and conda package. I believe everyone already has a hard copy of the executable (e.g. validate_magetab.pl) on their local machine. The config change, is the option to solve this on the server side, not client side (so to speak). But if you are not happy with this, I don't know what else is reasonable to do. I don't really want the change to go to validate_magetab.pl on atlas-prod code (as there are already so many other copies of the script).

Reply:
My second response was mostly for illustration purposes, I meant to follow with:

> lets just change the default behaviour and merge this as proposed.